src/lib/test-outbound/outbound-initiator.js

```diff
@@ -183,36 +183,20 @@ const terminateOutbound = (traceID) => {
 
 const processTestCase = async (testCase, traceID, inputValues, variableData, dfspId, globalConfig) => {
   const tracing = getTracing(traceID)
+  const apiDefinitions = openApiDefinitionsModel.getApiDefinitions()
 
-  // Load the requests array into an object by the request id to access a particular object faster
-  const requestsObj = {}
-  // Store the request ids into a new array
-  const templateIDArr = []
-  for (const i in testCase.requests) {
-    requestsObj[testCase.requests[i].id] = testCase.requests[i]
-    templateIDArr.push(testCase.requests[i].id)
-  }
-  // Sort the request ids array
-  templateIDArr.sort((a, b) => {
-    return a > b
-  })
-
-  const apiDefinitions = await openApiDefinitionsModel.getApiDefinitions()
-  // Iterate the request ID array
-  for (const i in templateIDArr) {
+  const sortedRequests = [ ...testCase.requests ].sort((a, b) => a.id - b.id)
```

Comment:
I am modifying the request object references and using the original template object at the end of test case execution for generating reports and sending final notifications to clients. I suspect cloning testCase.requests here may break some functionality.

Reply:
Okay, after a relatively difficult search, it looks like `setResponse` mutates the request object. I wonder if we could in fact not sort the array at all. Note that the original code was in fact not sorting the array at all:
```bash
> [1,6,3,5,4].sort((a, b) => a > b)
[ 1, 6, 3, 5, 4 ]
```

Reply:
Hmm, I think we can remove the sorting logic in that case. It seems we shouldn't sort based on the ID. The requests array should be executed in array order. Those sorting lines might be left from beginning some how and doesn't seem to have any effect.